TypedDict is not imported from typing_extensions in Python 3.14

### What happened?

Updating my project to test on Python 3.14, altair starts raising errors on import:

```
    import altair as alt
.tox/py314-pre/lib/python3.14/site-packages/altair/__init__.py:649: in <module>
    from altair.vegalite import *
.tox/py314-pre/lib/python3.14/site-packages/altair/vegalite/__init__.py:2: in <module>
    from .v5 import *
.tox/py314-pre/lib/python3.14/site-packages/altair/vegalite/v5/__init__.py:3: in <module>
    from altair.vegalite.v5 import api, compiler, schema
.tox/py314-pre/lib/python3.14/site-packages/altair/vegalite/v5/api.py:19: in <module>
    from altair import theme, utils
.tox/py314-pre/lib/python3.14/site-packages/altair/theme.py:9: in <module>
    from altair.vegalite.v5.schema._config import (
.tox/py314-pre/lib/python3.14/site-packages/altair/vegalite/v5/schema/_config.py:6491: in <module>
    class StepKwds(TypedDict, closed=True, total=False):  # type: ignore[call-arg]
E   TypeError: _TypedDictMeta.__new__() got an unexpected keyword argument 'closed'
```

It looks like `TypedDict` is imported from `typing_extensions` for Python<3.14, but from `typing` in Python 3.14+. I'm not 100% sure of the history of the [PEP 728](https://peps.python.org/pep-0728/), but it appears that it is only scheduled to be implemented in Python 3.15.

This is a redux of:

* #3684
* #3709

### What would you like to happen instead?

_No response_

### Which version of Altair are you using?

latest

--- a/altair/datasets/_typing.py
+++ b/altair/datasets/_typing.py
@@ -6,7 +6,7 @@ from __future__ import annotations
 import sys
 from typing import Literal
 
-if sys.version_info >= (3, 14):
+if sys.version_info >= (3, 15):
     from typing import TypedDict
 else:
     from typing_extensions import TypedDict

--- a/altair/vegalite/v6/api.py
+++ b/altair/vegalite/v6/api.py
@@ -38,7 +38,7 @@ from .schema.core import (
     VariableParameter,
 )
 
-if sys.version_info >= (3, 14):
+if sys.version_info >= (3, 15):
     from typing import TypedDict
 else:
     from typing_extensions import TypedDict

--- a/altair/vegalite/v6/schema/_config.py
+++ b/altair/vegalite/v6/schema/_config.py
@@ -15,7 +15,7 @@ if TYPE_CHECKING:
     from ._typing import *  # noqa: F403
 
 
-if sys.version_info >= (3, 14):
+if sys.version_info >= (3, 15):
     from typing import TypedDict
 else:
     from typing_extensions import TypedDict

--- a/altair/vegalite/v6/schema/_typing.py
+++ b/altair/vegalite/v6/schema/_typing.py
@@ -9,7 +9,7 @@ from collections.abc import Mapping, Sequence
 from datetime import date, datetime
 from typing import Annotated, Any, Generic, Literal, TypeVar, Union, get_args
 
-if sys.version_info >= (3, 14):  # https://peps.python.org/pep-0728/
+if sys.version_info >= (3, 15):  # https://peps.python.org/pep-0728/
     from typing import TypedDict
 else:
     from typing_extensions import TypedDict

--- a/tools/datasets/__init__.py
+++ b/tools/datasets/__init__.py
@@ -183,7 +183,7 @@ class Application:
             "from __future__ import annotations\n",
             "import sys",
             "from typing import Literal, TYPE_CHECKING",
-            utils.import_typing_extensions((3, 14), "TypedDict"),
+            utils.import_typing_extensions((3, 15), "TypedDict"),
             utils.import_typing_extensions((3, 11), "LiteralString"),
             utils.import_typing_extensions((3, 10), "TypeAlias"),
             "\n",

--- a/tools/datasets/models.py
+++ b/tools/datasets/models.py
@@ -6,7 +6,7 @@ import sys
 from collections.abc import Mapping, Sequence
 from typing import TYPE_CHECKING, Literal
 
-if sys.version_info >= (3, 14):
+if sys.version_info >= (3, 15):
     from typing import TypedDict
 else:
     from typing_extensions import TypedDict

--- a/tools/generate_schema_wrapper.py
+++ b/tools/generate_schema_wrapper.py
@@ -15,7 +15,7 @@ from pathlib import Path
 from typing import TYPE_CHECKING, Any, Final, Generic, Literal, TypeVar
 from urllib import request
 
-if sys.version_info >= (3, 14):
+if sys.version_info >= (3, 15):
     from typing import TypedDict
 else:
     from typing_extensions import TypedDict
@@ -1273,7 +1273,7 @@ def vegalite_main(skip_download: bool = False) -> None:
             "from collections.abc import Sequence",
             "from ._typing import * # noqa: F403",
         ),
-        import_typing_extensions((3, 14), "TypedDict"),
+        import_typing_extensions((3, 15), "TypedDict"),
         "\n\n",
         *generate_config_typed_dicts(schemafile),
     ]

--- a/tools/schemapi/utils.py
+++ b/tools/schemapi/utils.py
@@ -128,7 +128,7 @@ class _TypeAliasTracer:
             "from typing import Annotated, Any, Generic, Literal, TypeVar, Union, get_args",
             "import re",
             import_typing_extensions(
-                (3, 14), "TypedDict", reason="https://peps.python.org/pep-0728/"
+                (3, 15), "TypedDict", reason="https://peps.python.org/pep-0728/"
             ),
             import_typing_extensions((3, 13), "TypeIs"),
             import_typing_extensions((3, 12), "TypeAliasType"),

